refract-spec.md

```diff
@@ -137,9 +133,419 @@ This is how it would expressed in the compact version.
+      "content": "foo"
+    },
+    {
+      "element": "number",
+      "attributes": {},
+      "content": 400
+    },
+    {
+      "element": "boolean",
+      "attributes": {},
+      "content": true
+    }
+  ]
+}
+```
+
+### Property Element (Element)
+
+A `Property Type` allows for values that can be used in `Object Element` elements.
+
+#### Properties
+- `element`: property (string, fixed)
+- `attributes`
+    - `name` (string, required) - Name of property
+- `content` (Refract Element, required) - Any element that extends Refract Element
+
+#### Examples
+
+To see how this element is used, view the `Object Element` definition.
+
+### Object Element (Element)
+
+A `Object Element` allows for values that are JSON boolean.
+
+#### Properties
+- `element`: object (string, fixed)
+- `content` (array[Property Element], required)
+
+#### Examples
+
+In JSON, an example of an object is:
+
+```json
+{
+  "foo": "bar"
+}
+```
+
+In Refract, this object is expanded to a `Object Element`.
+
+```json
+{
+  "element": "object",
+  "attributes": {},
+  "content": [
+    {
+      "element": "property",
+      "attributes": {
+        "name": "foo"
+      },
+      "content": {
+        "element": "string",
+        "attributes": {},
+        "content": "bar"
+      }
+    }
+  ]
+}
+```
+
+## Referencing Elements
+
+The referencing elements are provided as part of the base specification for the purpose of identifying and referencing elements.
+
+### ID (Element)
+
+The `id` element is a way to identify an element in a document. IDs MUST be unique throughout a document.
+
+#### Example
+
+```json
+{
+  "element": "string",
+  "attributes": {
+    "id": "foo"
+  },
+  "content": "bar"
+}
+```
+
+### Ref (Element)
+
+The `ref` element can be used to reference elements in remove documents or elements in the local document.
+
+1. If the content is an absolute URL with a fragment identifier, the fragment identifer references the ID in the corresponding document.
+1. If the content is a relative fragment identifier, it references an element found in the local document
+1. If there is no fragment identifier, the reference is to the entire document.
+1. If there are multiple elements in a document with the same ID, the first element with the matching value SHOULD be used.
+
+The `ref` element transcludes the contents of the element into the document in which it is referenced.
+
+#### Properties
+
+- `element` ref (string, fixed)
+- `content` (string) - A URL
+
+#### Examples
+
+Elements can be referenced in remove or local documents.
+
+##### Referencing Remote Element
+
+```json
+{
+  "element": "ref",
+  "content": "http://example.com/document#foo"
+}
+```
+
+##### Referencing Local Elements
+
+```json
+{
+  "element": "ref",
+  "content": "#foo"
+}
+```
+
+## Namespacing
+
+Namespaces provide a way to include element definitions and constraints into another document. Namespaces do not include the elements and the data in the elements.
```

Comment:
I think you need to discuss the scope of the namespace being the element it is introduced in and all child elements, etc.